### DISCUSSION TITLE
Self-paced listening with picture on the screen

### DISCUSSION THREAD

**Discussion by ghost** [jspsych/jsPsych#769] [Q&A] [answered]
Hi all,

Hope you are well. I am new to JsPsych and am designing a self-paced listening task with picture verification. So the participants will see several pictures on the screen while hearing audio segments. Each picture has 6 audio segments. What I have got now seems fail to do so. Could you please let me know if this is viable and how should I make some changes to the code.

```javascript
var fixation = {
 type: 'html-keyboard-response',
 stimulus: '<div style="font-size:60px;">+</div>',
 choices: jsPsych.NO_KEYS,
 trial_duration: 1000,
}
singletrial.push(fixation)


let audio_stim = [`sound/1.wav`, `sound/2.wav`, `sound/3.wav`, etc.]
let image_file =['img/Bite1.jpg', etc.]
for (const values of audio_stim) 

 var trial = {
    type: 'audio-keyboard-response',
    stimulus: values,
    prompt: '<img src= "'+image_file+'">',
   };
   singletrial.push(trial);
}

var test_procedure = {
 timeline: singletrial,
//timeline_variables: trial
}

timeline.push(test_procedure);
//timeline.push(singletrial);
```

**Comment by ghost** [comment]
<img width="688" alt="WechatIMG979" src="https://user-images.githubusercontent.com/67168823/85176317-77d94e00-b271-11ea-9dc3-ce4a812ec81d.png">

**Comment by vijaymarupudi** [comment, marked as answer]
Can you check for errors in your browser's console?

https://developer.mozilla.org/en-US/docs/Tools/Web_Console\
https://developers.google.com/web/tools/chrome-devtools/console

**Reply by ghost** [reply]
Hi,

Thank you for your reply. There is actually no error. But only two audio segments pop up, although I entered three in the variable. Also, my utmost concern is that I need the same picture to be present on the screen for six different audio segments, consisting a trial. And a total of 48 trials need to be played. So what I can do if I manage to make it possible to maintain a picture on the screen while the sound files are changing, I might need to push each trial separately. Is there anyway to avoid this repetitiveness?

Thank you!

**Reply by jodeleeuw** [reply]
Check out the Timeline Variables feature. I think it will solve the repetitiveness problems.

* [Docs](https://www.jspsych.org/overview/timeline/#timeline-variables)
* [Tutorial Part 5: Timeline Variables](https://www.jspsych.org/tutorials/rt-task/#part-5-timeline-variables)

**Reply by ghost** [reply]
Hi Thank you! I have looked at it, but I still don't really get it as how to make it possible to have each picture having six audio segments with it. I suppose it should be like this :
`type: 'html-keyboard-response',
            stimulus: function(){
                var html="<img src='"+jsPsych.timelineVariable('picture', true)+"'>";
                html += "<p>"+jsPsych.timelineVariable('segment1', true)+">";
               html += "<p>"+jsPsych.timelineVariable('segment2', true)+">"
               html += "<p>"+jsPsych.timelineVariable('segment3', true)+">" 
              html += "<p>"+jsPsych.timelineVariable('segment4', true)+">"
              html += "<p>"+jsPsych.timelineVariable('segment5, true)+">"
             html += "<p>"+jsPsych.timelineVariable('segment6', true)+"</p>"
                return html;
            },          
            choices: jsPsych.NO_KEYS,
            trial_duration: 2500
        }
    ],
    timeline_variables: [
        {picture: 'person-1.jpg', segment1: 'audio1', segment2 :'audio2',etc. },
        { picture: 'person-2.jpg', segment1: 'audio7', segment2 :'audio8',etc.  },
        { picture: 'person-3.jpg', segment1: 'audio13', segment2 :'audio14',etc. },
        { picture: 'person-4.jpg', segment1: 'audio19', segment2 :'audio20',etc.  }
    ]`


But it doesn't work...
